rust/shalom_dart_codegen/dart_tests/test/operation_scalar_arguments/__graphql__/GetProductDetails.shalom.dart

```diff
@@ -0,0 +1,243 @@
+    }
+
+    return GetProductDetailsResponse(product: product_value);
+  }
+
+  @override
+  bool operator ==(Object other) {
+    return identical(this, other) ||
+        (other is GetProductDetailsResponse && other.product == product);
+  }
+
+  @override
+  int get hashCode => product.hashCode;
+
+  JsonObject toJson() {
+    return {'product': product?.toJson()};
+  }
+}
+
+// ------------ OBJECT DEFINITIONS -------------
+
+class GetProductDetails_product {
+  /// class members
+
+  final String id;
+
+  final String name;
+
+  final double price;
+
+  final double? discountedPrice;
+
+  // keywordargs constructor
+  GetProductDetails_product({
+    required this.id,
+    required this.name,
+    required this.price,
+
+    this.discountedPrice,
+  });
+  static GetProductDetails_product fromJson(JsonObject data) {
+    final String id_value = data['id'];
+
+    final String name_value = data['name'];
+
+    final double price_value = data['price'];
+
+    final double? discountedPrice_value = data['discountedPrice'];
+
+    return GetProductDetails_product(
+      id: id_value,
+
+      name: name_value,
+
+      price: price_value,
+
+      discountedPrice: discountedPrice_value,
+    );
+  }
+
+  GetProductDetails_product updateWithJson(JsonObject data) {
+    final String id_value;
+    if (data.containsKey('id')) {
+      id_value = data['id'];
+    } else {
+      id_value = id;
+    }
+
+    final String name_value;
+    if (data.containsKey('name')) {
+      name_value = data['name'];
+    } else {
+      name_value = name;
+    }
+
+    final double price_value;
+    if (data.containsKey('price')) {
+      price_value = data['price'];
+    } else {
+      price_value = price;
+    }
+
+    final double? discountedPrice_value;
+    if (data.containsKey('discountedPrice')) {
+      discountedPrice_value = data['discountedPrice'];
+    } else {
+      discountedPrice_value = discountedPrice;
+    }
+
+    return GetProductDetails_product(
+      id: id_value,
+
+      name: name_value,
+
+      price: price_value,
+
+      discountedPrice: discountedPrice_value,
+    );
+  }
+
+  @override
+  bool operator ==(Object other) {
+    return identical(this, other) ||
+        (other is GetProductDetails_product &&
+            other.id == id &&
+            other.name == name &&
+            other.price == price &&
+            other.discountedPrice == discountedPrice);
+  }
+
+  @override
+  int get hashCode => Object.hashAll([id, name, price, discountedPrice]);
+
+  JsonObject toJson() {
+    return {
+      'id': id,
+
+      'name': name,
+
+      'price': price,
+
+      'discountedPrice': discountedPrice,
+    };
+  }
+}
+
+// ------------ END OBJECT DEFINITIONS -------------
+
+class RequestGetProductDetails extends Requestable {
+  final GetProductDetailsResponse operation;
+  final GetProductDetailsVariables variables;
+
+  RequestGetProductDetails({required this.operation, required this.variables});
```

Comment:
operation is statically known at codgen time thus no need for it to be in the constructor AFAICT